app/models/calnet_ldap/person.rb

```diff
@@ -0,0 +1,75 @@
+module CalnetLdap
+  # https://calnetweb.berkeley.edu/calnet-technologists/ldap-directory-service/how-ldap-organized/people-ou
+  class Person
+    def self.get(user)
+      if net_ldap_entry = CalnetLdap::Client.new.search_by_uid(user.uid)
+        return self.new(net_ldap_entry)
+      end
+      nil
+    end
+
+    def initialize(net_ldap_entry)
+      if net_ldap_entry.respond_to?(:attribute_names)
```

Comment:
Is there a reason you're checking if the incoming object responds_to? :attribute_names. This would introduce a silent failure if it doesn't. An `NoMethodError: undefined method 'attribute_names'` would actually be a useful error message in that case. Also I would consider it the responsibility of the caller to initialize the class with an appropriate object.